src/Http/Services/PageService.php

```diff
@@ -29,4 +30,28 @@ public function getBySlug(string $slug): Page
     {
         return $this->repository->getBySlug($slug);
     }
+
+    /**
+     * @param string $slug
+     * @param string $title
+     * @param string $content
+     * @param int $userId
+     * @return Page
+     * @throws PageAlreadyExistsException
+     */
+    public function insert(string $slug, string $title, string $content, int $userId): Page
+    {
+        /** @var Page $page */
+        $page = Page::factory()->newModel([
+            'slug'=>$slug,
+            'title'=>$title,
+            'author_id'=>$userId,
+            'content'=>$content,
+        ]);
+        $this->repository->insert($page);
```

Comment:
I'm trying to understand this fragment, but I'm not sure when `repository->insert` would not save Page properly? It calls createUsingModel from Core/BaseRepository class, which should save model, unless there is some missing nonnullable data or some unique index conflict, but it will probably throw Exception then anyway...

And why do you throw "PageAlreadyExists" when the page is not saved properly?

Reply:
I've updated the method of checking by trying to find the model first.
Is it ok now?